backend/database.js

```diff
@@ -172,6 +173,51 @@ export default (sbp('sbp/selectors/register', {
   }
 }): any)
 
+if (persistence === 'fs') {
+  sbp('sbp/selectors/register', {
+    'backend/db/readString': async function (key: string): Promise<string | null> {
+      const bufferOrError = await sbp('backend/db/readFile', key)
+      if (Boom.isBoom(bufferOrError)) {
+        return null
+      }
+      return bufferOrError.toString('utf8')
+    },
+    'backend/db/writeString': async function (key: string, value: string): Promise<void> {
+      // eslint-disable-next-line no-useless-catch
+      try {
+        return await sbp('backend/db/writeFile', key, value)
+      } catch (err) {
+        throw err
+      }
+    }
+  })
+} else if (persistence === 'sqlite') {
+  sbp('sbp/selectors/register', {
+    'backend/db/readString': function (key: string): Promise<string> {
+      return new Promise((resolve, reject) => {
+        db.get('SELECT value FROM Strings WHERE key = ?', [key], (err, row) => {
+          if (err) {
+            reject(err)
+          } else {
+            resolve(row?.value)
+          }
+        })
+      })
+    },
+    'backend/db/writeString': function (key, value) {
+      return new Promise((resolve, reject) => {
+        db.run('REPLACE INTO Strings(key, value) VALUES(?, ?)', [key, value], (err) => {
```

Comment:
Why do you think `REPLACE` should be used here instead of `INSERT`?
Do you think it would be necessary to update string?
Couldn't be there any security problem?

Reply:
Wouldn't `INSERT` fail if an entry with the given key already exists?

Reply:
When would it be happening to add entry with the already existing key? I don't think there will be.

Reply:
For example, when updating the current HEAD for a given contract ID key. I think this can happen many times for any contract.

Reply:
I am not sure why I missed this. Hahaha. Totally agreed.